src/collapse/perf.rs

```diff
@@ -233,36 +334,41 @@ impl Folder {
                 if event.ends_with(':') {
                     let event = &event[..(event.len() - 1)];
 
-                    if let Some(ref filter) = self.opt.event_filter {
-                        if event != filter {
-                            if let EventFilterState::Defaulted = self.event_filtering {
-                                // only print this warning if necessary:
-                                // when we defaulted and there was
-                                // multiple event types.
+                    match self.event_filter_state {
```

Comment:
This change could probably be in its own commit.

Reply:
This gets back to the issue above regarding clearly differentiating between "configuration" and "state". In the master branch, the `event_filter` field of `Options` was being used as both "configuration" and "state," which took me a while to figure out and is also why I messed with this part of the code.  We should make this distinction clear for future folks.  As mentioned in the comment above, I'll clean up so as to make a clear distinction and also brings back the `opt` field on `Folder`.

Reply:
What I'll probably do is put the `event_filter: Option<String>` field back on `Options`; so that the initial option the user passed in remains available and known. Then I'll keep the `event_filter_state` field on the `Folder` struct as it is in this branch. Feel free to wait until you see a next revision in order to comment, as my prose may not be super clear.